pkg/kv/kvclient/rangefeed/rangefeed.go

```diff
@@ -280,19 +280,16 @@ func (f *RangeFeed) run(ctx context.Context, frontier *span.Frontier) {
 
 		start := timeutil.Now()
 
-		// Note that the below channel send will not block forever because
-		// processEvents will wait for the worker to send. RunWorker is safe here
-		// because processEvents is guaranteed to consume the error before
-		// returning.
-		if err := f.stopper.RunAsyncTask(ctx, "rangefeed", func(ctx context.Context) {
-			errCh <- f.client.RangeFeed(ctx, f.spans, ts, f.withDiff, eventCh)
-		}); err != nil {
-			log.VEventf(ctx, 1, "exiting rangefeed due to stopper")
-			return
+		rangeFeedTask := func(ctx context.Context) error {
+			return f.client.RangeFeed(ctx, f.spans, ts, f.withDiff, eventCh)
+		}
+		processEventsTask := func(ctx context.Context) error {
+			return f.processEvents(ctx, frontier, eventCh)
```

Comment:
That's nice.  Whatever run async task did, this function (run) already runs under that -- so spans, ctx should already be setup.

Reply:
You can probably move those definition outside the loop.

Reply:
>Whatever run async task did, this function (run) already runs under that -- so spans, ctx should already be setup.

Not entirely -- `RunAsyncTask` will fork a new tracing span for the rangefeed goroutine. Without that, these two goroutines will be sharing the same tracing span.

> You can probably move those definition outside the loop.

The rangefeed needs to close over the `ts` variable which is scoped to the loop. I don't think it's safe to share `frontier` between the goroutines.

Reply:
Ahh.. I missed the ts arg -- you're right.  But the frontier is not shared -- we wait for the previous go routine to finish before we start next one... Anyway, it's probably clearer to keep things as they are.